timemachine/md/minimizer.py

```diff
@@ -145,12 +147,18 @@ def minimize_host_4d(mols, host_system, host_coords, ff, box, mol_coords=None) -
     x0 = combined_coords
     v0 = np.zeros_like(x0)
 
-    x0 = fire_minimize(x0, u_impls, box, np.ones(50))
-    # context components: positions, velocities, box, integrator, energy fxns
-    ctxt = custom_ops.Context(x0, v0, box, intg, u_impls)
-    _, xs, _ = ctxt.multiple_steps(np.linspace(1.0, 0, 1000))
+    u_impls = bind_potentials(parameterize_system(hgt, ff, 1.0))
+    x = fire_minimize(x0, u_impls, box, np.ones(50))
+
+    for lamb in np.linspace(1.0, 0, 50):
+        u_impls = bind_potentials(parameterize_system(hgt, ff, lamb))
+        # context components: positions, velocities, box, integrator, energy fxns
+        ctxt = custom_ops.Context(x, v0, box, intg, u_impls)
+        _, xs, _ = ctxt.multiple_steps(lamb * np.ones(50))
+        x = xs[-1]
```

Comment:
Note: this reduces the number of lambda windows used for minimization from 1000 down to 50 (while increasing the number of steps per window from 1 to 50). This is probably less efficient, but since we currently need to construct a new bound potential for each lambda value, it was necessary to reduce the number of lambda windows.

Aside: it might make sense for a future PR to add an interface to `Context` that allows passing unbound potentials and parameters separately? That way we could at least avoid the cost of constructing a new potential object on each iteration here.

Reply:
> Aside: it might make sense for a future PR to add an interface to Context that allows passing unbound potentials and parameters separately? That way we could at least avoid the cost of constructing a new potential object on each iteration here.

I think that's the cleanest way.